arelle/plugin/validate/ESEF_2022/Util.py

```diff
@@ -167,3 +169,17 @@ def checkForMultiLangDuplicates(modelXbrl: ModelXbrl) -> None:
                         "Inconsistent duplicate non-numeric facts SHOULD NOT appear in the content of an inline XBRL document. "
                         "%(fact)s that was used more than once in contexts equivalent to %(contextID)s, with different values but same language (%(language)s).",
                         modelObject=fList, fact=fList[0].qname, contextID=fList[0].contextID, language=fList[0].xmlLang)
+
+def isAnchoredToNotes(child: ModelConcept, relSet: ModelRelationshipSet, _visited: set[ModelConcept]) -> bool:
```

Comment:
This doesn't appear to restrict the taxonomy tree walk to the link roles of the original child concept.

Consider the following taxonomy structure:

#### Link role 1
```
NoneESEFNotesStatementConcept -> ParentConcept -> ChildConcept
```

#### Link role 2
```
ESEFNotesStatementConcept -> ParentConcept -> SomeOtherChildConcept
```

If a `ModelRelationshipSet` with parent child relationships for the entire taxonomy is passed in this function will incorrectly report `ChildConcept` as a child of notes to the financial statements because the search will jump to link role 2 when `ParentConcept` is checked.

You can limit the link roles searched by restricting the relationship set that's passed in.
```python
conceptRels = parentChildRelSet.toModelObject(modelConcept)
conceptLinkroles = tuple(set(rel.linkrole for rel in conceptRels))
conceptLinkroleRestrictedRelSet = val.modelXbrl.relationshipSet(XbrlConst.parentChild, conceptLinkroles)
if not isAnchoredToNotes(modelConcept, conceptLinkroleRestrictedRelSet, set()):
```

Reply:
Requested change has been performed